src/isar_turtlebot/robotinterface.py

```diff
@@ -39,21 +41,61 @@ def __init__(self):
         self.bridge: RosBridge = RosBridge()
 
     def schedule_step(self, step: Step) -> Tuple[bool, Optional[Any], Optional[Joints]]:
-        self._publish_task(step=step)
-        previous_run_id: str = self._get_run_id()
-        run_id: str = self._wait_for_updated_task(previous_run_id=previous_run_id)
+        run_id: str = self._publish_task(step=step)
         return True, run_id, None
 
     def _publish_task(self, step: Step) -> None:
         if isinstance(step, DriveToPose):
+            self.current_task = "navigation"
+            previous_run_id: str = self._get_run_id()
             self._publish_navigation_task(pose=step.pose)
+            run_id: str = self._wait_for_updated_task(previous_run_id=previous_run_id)
 
+            return run_id
         elif isinstance(step, TakeImage):
-            pose: Pose = get_inspection_pose(
-                current_pose=self.robot_pose(), target=step.target
-            )
-            self._publish_navigation_task(pose=pose)
-            self.bridge.visual_inspection.take_image()
+            self.current_task = "inspection"
+            run_id: str = self._publish_inspection_task(target=step.target)
+            try:
+                self._do_inspection_task()
+            except TimeoutError as e:
+                self.logger.error(e)
+                self.current_task = None
+
+            return run_id
+
+    def _do_inspection_task(self):
+        start_time: float = time.time()
+        timeout: int = 20
+        while self._navigation_status() is not TurtlebotStatus.Succeeded:
+            time.sleep(0.1)
+            execution_time: float = time.time() - start_time
+            if execution_time > timeout:
+                self.inspection_status = TurtlebotStatus.Failure
+                raise TimeoutError(
+                    f"Drive to inspection pose task for TurtleBot3 timed out. Run ID: {self._get_run_id()}"
+                )
+
+        self.bridge.visual_inspection.take_image()
+        while not self.bridge.visual_inspection.current_filename.is_file():
```

Comment:
Could this be an own function in the visual_inspection_topic, instead of the robot interface calling functions on specific variables?